plugins/ubiquiti/unifi_api

```diff
@@ -0,0 +1,1138 @@
+
+    env.enable_xfer_port yes
+    # Show transfer statistics on switch ports
+    env.enable_detail_xfer_port yes
+    # Show detailed graphs per switch port
+    env.hide_empty_xfer_port yes
+    # Hide ports that have no link
+
+    env.enable_xfer_device yes
+    # Show transfer statistics per device
+    env.enable_detail_xfer_device yes
+    # Show detailed graphs for each device
+
+    env.enable_xfer_network yes
+    # Show transfer statistics per named network
+    env.enable_detail_xfer_network yes
+    # Show detailed graphs for each named network
+
+    env.enable_xfer_radio yes
+    # Show transfer statistics per radio
+    env.enable_detail_xfer_radio yes
+    # Show detailed graphs for each radio
+
+
+=head1 CAPABILITIES
+
+This plugin requires multigraph.
+
+This plugin supports L<DIRTYCONFIG|http://guide.munin-monitoring.org/en/latest/plugin/protocol-dirtyconfig.html>
+
+=head1 DEPENDENCIES
+
+=over
+
+=item WWW::Curl::Easy
+
+Perl extension interface for libcurl
+
+=item JSON
+
+ JSON (JavaScript Object Notation) encoder/decoder
+
+=back
+
+=head1 PERFORMANCE CONCERNS
+
+The main performance concern on this is the huge number of graphs that may be
+generated.  Using the cron version of munin-graph may hurt a lot.
+
+A bit of a case study:
+
+                 | My Site  | UBNT Demo
+---------------------------------------
+Devices          |       8  |     126
+AP's             |       4  |     118
+24xSwitch        |       1  |       5
+8xSwitch         |       2  |       2
+Output Bytes     |  64,262  | 431,434
+Output Lines     |   1,761  |  14,586
+Output Graphs    |      77  |     530
+
+So, just note that the growth in the amount of graphed date can be extream.
+
+
+=head1 LICENSE
+
+Copyright (C) 2018 J.T.Sage (jtsage@gmail.com)
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see L<http://www.gnu.org/licenses/>.
+
+=head1 VERSION
+
+Most recent version available from 
+
+L<https://github.com/jtsage/unifi-munin-plugins>
+
+=head1 MAGICK MARKERS
+
+  #%# family=manual
+  #%# capabilities=autoconf
+
+=cut
+
+use warnings;
+use strict;
+use utf8;
+
+# Check dependancies
+my @errorCode;
+my $me = (split '/', $0)[-1];
+
+if (! eval {require JSON; JSON->import(); 1; } ) {
+	push @errorCode, "JSON module not found";
+}
+if (! eval {require WWW::Curl::Easy; 1;} ) {
+	push @errorCode, "WWW::Curl::Easy module not found";
+}
+
+# Provide autoconf info
+if ( defined($ARGV[0]) && $ARGV[0] eq "autoconf" ) {
```

Comment:
It looks like this plugin cannot work out of the box (due the unknown URL of the API provider). In this case you should just omit the `autoconf` capability.
Or is the default `api_url` a good guess, that could work sometimes?

Reply:
I really did it more as a way to check dependancies - then I backported that check to *always* do it, even when config or fetch are run - I'll drop the autoconf bit - the defaults *could* work, but they really, really shouldn't.  URL might be fine - username/password sure shouldn't be.